coded_tools/now_agents/nowagent_api_get_agents.py

```diff
@@ -0,0 +1,125 @@
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san-studio SDK Software in commercial settings.
+#
+import os
+from typing import Any, Dict
+
+import requests
+from neuro_san.interfaces.coded_tool import CodedTool
+
+
+class NowAgentAPIGetAgents(CodedTool):
+    """
+    A tool to discover and retrieve available ServiceNow AI agents from a ServiceNow instance.
+
+    This tool queries the sn_aia_agent table in ServiceNow to get a list of active AI agents
+    along with their descriptions and system IDs for further interaction.
+
+    Example usage: See tests in the now_agents module.
+    """
+
+    def __init__(self):
+        """
+        Constructs a NowAgentAPIGetAgents object.
+
+        No initialization parameters required. Configuration is handled through environment variables.
+        """
+
+    def invoke(self, args: Dict[str, Any], sly_data: Dict[str, Any]) -> str:
+        """
+        Discovers and retrieves available ServiceNow AI agents from the configured instance.
+
+        This method queries the ServiceNow instance's sn_aia_agent table to get a list of
+        AI agents that match the configured query criteria (typically active agents).
+
+        Args:
+            args: Dictionary containing the inquiry parameters (not used for agent discovery)
+            sly_data: Dictionary for session data management (not used for agent discovery)
+
+        Returns:
+            dict: ServiceNow API response containing the list of agents with their details.
+                  Response structure includes:
+                  - result: List of agent dictionaries with 'description', 'name', and 'sys_id' fields
+
+        Raises:
+            SystemExit: If the ServiceNow API returns a non-200 status code
+        """
+        # Parse the arguments
+        servicenow_url: str = self._get_env_variable("SERVICENOW_INSTANCE_URL")
+        servicenow_get_agents_query: str = self._get_env_variable("SERVICENOW_GET_AGENTS_QUERY")
+        servicenow_user: str = self._get_env_variable("SERVICENOW_USER")
+        servicenow_pwd: str = self._get_env_variable("SERVICENOW_PWD")
+        print(f"ServiceNow URL: {servicenow_url}")
+        print(f"user:{servicenow_user}")
+        print(f"pwd:{servicenow_pwd}")
+
+        tool_name = self.__class__.__name__
+        print(f"========== Calling {tool_name} ==========")
+
+        # Build the ServiceNow API URL for agent discovery
+        base_url = f"{servicenow_url}api/now/table/sn_aia_agent"
+        query_params = f"sysparm_query={servicenow_get_agents_query}"
+        field_params = "sysparm_fields=description%2Cname%2Csys_id"
+        url = f"{base_url}?{query_params}&{field_params}"
+
+        # Set proper headers
+        headers = {"Content-Type": "application/json", "Accept": "application/json"}
+
+        # Execute the HTTP request
+        response = requests.get(url, auth=(servicenow_user, servicenow_pwd), headers=headers)
```

Comment:
This call can take some time I guess. An `await` here and moving the implementation to `async_invoke` could be a nice enhancement in the future.
Same thing for the 2 other CodedTools